pkg/webhook/genericcloudevent.go

```diff
@@ -0,0 +1,114 @@
+package webhook
```

Comment:
Would it be prudent to create a "Generic Webhook" interface that both the simple event and cloud event webhooks could then implement?  Arguments for this would be if a non-trivial amount of code (including the test code) would be consolidated in doing so and/or if additional generic webhook variants might be added in the future... 

Definitely defer to the author on this idea -- also, even if it might gain momentum, can surely be in the form of a follow-up refactor ticket.

Reply:
That's a great idea and definitely something I'm considering.
The only thing that pinches me if we go down this path is that I'm not sure whether in the future we would like to have any special treatment for a CloudEvent since it carries a [context](https://github.com/cloudevents/spec/blob/v0.2/spec.md#context-attributes) and we may want to do different things depending on its content. E.g. [source](https://github.com/cloudevents/spec/blob/v0.2/spec.md#source) might be an absolute path, a relative path, a urn, an email, etc. and we could want to have the CloudEvent handler take different actions depending on it.
On the contrary, a SimpleEvent has no pre-defined schema so I doubt we'll want to do anything more rather than just checking if there are ref/commit values present.
What do you think?

Reply:
I think for for we can keep it this way, and as it becomes more clear if the implementations for later versions of CloudEvents and our simple event converge or diverge, we can re-evaluate.

Reply:
Thanks @dgkanatsios, I'd figured you had considered both approaches.  The approach in this PR definitely seems the sensible path given the points you've mentioned.  In agreement with @radu-matei above -- makes sense to revisit at a later date.